modules/utils/tests/file_tests.cpp

```diff
@@ -0,0 +1,32 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include <gtest/gtest.h>
+
+#include "hephaestus/utils/filesystem/file.h"
+
+// NOLINTNEXTLINE(google-build-using-namespace)
+using namespace ::testing;
+
+namespace heph::utils::filesystem::tests {
+TEST(BinaryFile, ReadWrite) {
+  const auto path = ScopedPath::createFile();
+  const auto content = std::vector<std::byte>{ std::byte{ 0x01 }, std::byte{ 0x02 }, std::byte{ 0x03 } };
+
+  writeBufferToFile(path, { content.data(), content.size() });
+  const auto read_content = readBinaryFile(path);
+
+  EXPECT_EQ(content, read_content);
+}
+
+TEST(File, ReadWrite) {
```

Comment:
nit. The first param of TEST should be a collection tag for tests, e.g. TEST(HephFilesystem, ...).